### PR TITLE
add seloger.com scraper

### PR DESCRIPTION
**Pull request by mazen-r**

### CODE REVIEW

File: seloger-scraper/seloger.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    # select the script tag from the HTML
+    script_jsons = result.selector.xpath("//script[contains(., 'window[\"initialData\"]')]/text()").re(
+        r'JSON.parse\("(.+?)"\)'
+    )
+    # decode the JSON data
+    datasets = [json.loads(script_json.encode("utf-8").decode("unicode_escape")) for script_json in script_jsons]
+    cards = []
+    # validate the cards data to avoid advertisement cards
+    for card in datasets[0]["cards"]["list"]:
+        if len(card) > 12:
```
Comment (Granitosaurus) [inline seloger-scraper/seloger.py:36]: so the cards with <= 12 datapoint keys are ads? Maybe there's a better way to identify this? Seems a bit flimsy.

Comment (Granitosaurus) [inline seloger-scraper/seloger.py:36]: so the cards with <= 12 datapoint keys are ads? Maybe there's a better way to identify this? Seems a bit flimsy.

File: seloger-scraper/run.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+async def run():
+    # enable scrapfly cache for basic use
+    seloger.BASE_CONFIG["cache"] = True
+
+    print("running Seloger scrape and saving results to ./results directory")
+
+    search_data = await seloger.scrape_search(
+        url="https://www.seloger.com/immobilier/achat/immo-bordeaux-33/bien-appartement/",
+        scrape_all_pages=False,
+        max_pages=800,
```
Comment (Granitosaurus) [inline seloger-scraper/run.py:26]: we should limit this to fewer pages

File: seloger-scraper/seloger.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    search_data = search_page_result["results"]
+    # get the max search pages number
+    total_search_pages = _max_search_pages(search_page_result["search"])
+    # scrape a specfic amount of search pages
+    if scrape_all_pages == False and max_pages < total_search_pages:
+        total_pages = max_pages
+    # scrape all available pages in the search if scrape_all_pages = True or max_pages > total_search_pages
+    else:
+        total_pages = total_search_pages
+    print("-----------------", total_pages, "----------------")
```
Comment (Granitosaurus) [inline seloger-scraper/seloger.py:79]: print debug leftover

File: seloger-scraper/seloger.py [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+    property_data["agency"] = listing_data["agency"]
+    return property_data
+
+
+async def scrape_search(
+    url: str,
+    scrape_all_pages: bool,
+    max_pages: int = 10,
+) -> List[Dict]:
+    """scrape seloger search page"""
```
Comment (Granitosaurus) [inline seloger-scraper/seloger.py:64]: can we add exmaple search URL so people can understand the function faster? Same for `scrape_property`

**Review by Granitosaurus — COMMENTED** [review]
lgtm though left some minor comments

**Comment by mazen-r** [comment]
Done ✅
Also, the property page suddenly has become 404. I have updated the property page URL and added some nullable fields to allow some missing fields on the new property page.

**Pull request merged**